src/commitStrategy/BasicCommitStrategy.ts

```diff
@@ -0,0 +1,99 @@
+import CommitStrategy from './CommitStrategy';
+import IObjectMetadata from '../objects/IObjectMetadata';
+import Commit from '../commits/Commit';
+import CommitOperation from '../commits/CommitOperation';
+
+export const BASIC_COMMIT_STRATEGY = 'basic';
+
+/**
+ * BasicCommitStrategy implements the 'basic' commit strategy
+ */
+export default class BasicCommitStrategy extends CommitStrategy {
+  resolveObject (objectId: string, commits: Commit[]): {
+    /** Resolved object metadata */
+    metadata: IObjectMetadata,
+    /** Resolved object data */
+    data: any;
+  } {
+    // reduce to only those of the right object
+    const basicCommits = this.filterCommits(objectId, BASIC_COMMIT_STRATEGY, commits);
+    // get the latest commit
+    const latestCommit = BasicCommitStrategy.findLatestCommit(basicCommits);
+    const earliestCommit = BasicCommitStrategy.findCreateCommit(objectId, basicCommits);
+    const latestMeta = latestCommit.getHeaders();
+    const staticMetadata = earliestCommit ? earliestCommit.getHeaders() : {

```

Comment:
I would suggest not resolving the object (i.e. throw an error) if we don't have the create commit